modules/network/src/main/java/org/apache/ignite/internal/network/direct/DirectMessageWriter.java

```diff
@@ -54,12 +60,15 @@ public class DirectMessageWriter implements MessageWriter {
      */
     public DirectMessageWriter(MessageSerializationRegistry serializationRegistry, byte protoVer) {
         state = new DirectMessageState<>(StateItem.class, () -> new StateItem(createStream(serializationRegistry, protoVer)));
+
+        stateItem = state.item();
+        stream = stateItem.stream;
```

Comment:
Wow, is stateItem.stream noticeably slower than this.stream, to add extra field?

One might forget to change stream after changing item, maybe we should add assertion that stream equals to item().stream? Or I'm paranoid

Reply:
Adding assertions would make code slower when you enable assertions :)
No one will forget that, because methods that modify stream are pretty stable and we're not planning to add more of them